src/services/AudioManager.ts

```diff
@@ -0,0 +1,105 @@
+import BN from 'bn.js'
+import { put, call } from 'redux-saga/effects'
+
+import AudiusBackend from 'services/AudiusBackend'
+import { Name } from 'services/analytics'
+import walletClient from 'services/wallet-client/WalletClient'
+import { make } from 'store/analytics/actions'
+import { setState as setAudioManagerReduxState } from 'store/audio-manager/slice'
+import * as errorActions from 'store/errors/actions'
+import { Nullable } from 'utils/typeUtils'
+
+export enum AudioManagerState {
+  HasEthAudioInHedgehog = 'HAS_ETH_AUDIO_IN_HEDGEHOG',
+  HasRequestedMoveEthAudioInHedgehog = 'HAS_REQUESTED_MOVE_ETH_AUDIO_IN_HEDGEHOG',
+  TransferringEthAudioInHedgehogToSPLBankAccount = 'TRANSFERING_ETH_AUDIO_IN_HEDGEHOG_TO_SPL_BANK_ACCOUNT',
+  ErrorTransferringEthAudioInHedgehogToSPLBankAccount = 'ERROR_TRANSFERING_ETH_AUDIO_IN_HEDGEHOG_TO_SPL_BANK_ACCOUNT',
+  HasNoEthAudioInHedgehog = 'HAS_NO_ETH_AUDIO_IN_HEDGEHOG'
+}
+
+type AudioManagerProps = {
+  requestTransferAudioToWAudio: Generator<any, boolean, unknown>
+}
+
+class AudioManager {
+  state: Nullable<AudioManagerState>
+  requestTransferAudioToWAudio: Generator<any, boolean, unknown>
+  constructor({ requestTransferAudioToWAudio }: AudioManagerProps) {
+    this.state = null
+    this.requestTransferAudioToWAudio = requestTransferAudioToWAudio
+
+    this.getInitState = this.getInitState.bind(this)
+    this.updateState = this.updateState.bind(this)
+    this.updateStatePhase = this.updateStatePhase.bind(this)
+  }
+
+  *getInitState() {
+    // @ts-ignore
+    const user = yield call(AudiusBackend.getAccount)
+    if (user.balance === '0') {
+      yield call(
+        this.updateStatePhase,
+        AudioManagerState.HasNoEthAudioInHedgehog
+      )
+      this.state = AudioManagerState.HasNoEthAudioInHedgehog
+      return this.state
+    }
+    yield call(this.updateStatePhase, AudioManagerState.HasEthAudioInHedgehog)
+    return this.state
+  }
+
+  *updateState() {
+    if (this.state === AudioManagerState.HasEthAudioInHedgehog) {
```

Comment:
switch statement maybe?

Reply:
moved to helper function and put in switch